lib/dao.js

```diff
@@ -1550,15 +1571,18 @@ DataAccessObject._coerce = function(where) {
     // Handle logical operators
     if (p === 'and' || p === 'or' || p === 'nor') {
       var clauses = where[p];
-      if (Array.isArray(clauses)) {
-        for (var k = 0; k < clauses.length; k++) {
-          self._coerce(clauses[k]);
-        }
-      } else {
+      try {
+        clauses = coerceArray(clauses);
+      } catch (e) {
         err = new Error(g.f('The %s operator has invalid clauses %j', p, clauses));
```

Comment:
Please log the `e` error (at least via `debug`), so that users have means how to investigate what exactly is wrong with their clauses.